### PR TITLE
Seccomp lock personality

### PR DESCRIPTION
**Pull request by poettering**
A rebase of @topimiettinen's PR #6256, with some more additions. @topimiettinen please have a look!

### CODE REVIEW

File: man/systemd.exec.xml [hunk 1]
```diff
@@ -1650,6 +1650,18 @@
...
 
       <varlistentry>
+        <term><varname>LockPersonality=</varname></term>
+
+        <listitem><para>Locks down the <citerefentry
+        project='man-pages'><refentrytitle>personality</refentrytitle><manvolnum>2</manvolnum></citerefentry> system
+        call so that the kernel execution domain may not be changed from the default or the personality selected with
+        <varname>Personality=</varname> directive. This may be useful to improve security, because odd personality
+        emulations may be poorly tested and source of vulnerabilities. If running in user mode, or in system mode, but
+        without the <constant>CAP_SYS_ADMIN</constant> capability (e.g. setting <varname>User=</varname>),
```
Comment (boucman) [inline man/systemd.exec.xml:1664]: It took me three readings to understand that sentence... I'm not sure my rewording is better but...

if the service is run without CAP_SYS_ADMIN or as a non-root user,  NoNewPrivileges=yes is implied

I think removing the (e.g.) part makes thngs a bit clearer

Comment (topimiettinen) [reply man/systemd.exec.xml:1664]: This part was copied from elsewhere, the same applies to a few other locations.

Comment (yuwata) [reply man/systemd.exec.xml:1664]: > if the service is run without CAP_SYS_ADMIN or as a non-root user, NoNewPrivileges=yes is implied

I think this is not correct. If the service is run as a non-root user with CAP_SYS_ADMIN, then NoNewPrivileges= is not implied. I think @topimiettinen's original sentence is correct.

**Comment by yuwata** [comment]
xenial-i386 CI fails. test-execute:
```
Child 20234 (sh) died (code=exited, status=1/FAILURE)
exec-systemcallerrornumber.service: Child 20234 belongs to exec-systemcallerrornumber.service
exec-systemcallerrornumber.service: Main process exited, code=exited, status=1/FAILURE
exec-systemcallerrornumber.service: Changed start -> failed
exec-systemcallerrornumber.service: Unit entered failed state.
exec-systemcallerrornumber.service: Failed with result 'exit-code'.
exec-systemcallerrornumber.service: cgroup is empty
```
and similar failures. Also, test-seccomp:
```
Assertion 'personality(PER_LINUX | ADDR_NO_RANDOMIZE) == -1 && errno == EPERM' failed at ../src/test/test-seccomp.c:583, function test_lock_personality(). Aborting.
lockpersonalityseccomp terminated by signal ABRT.
Assertion 'wait_for_terminate_and_warn("lockpersonalityseccomp", pid, true) == EXIT_SUCCESS' failed at ../src/test/test-seccomp.c:599, function test_lock_personality(). Aborting.
FAIL: test-seccomp (code: 134)
```

File: man/systemd.exec.xml (hunk 1 above)
Comment (poettering) [reply man/systemd.exec.xml:1664]: yupp, @topimiettinen's wording is correct.

**Comment by poettering** [comment]
Force pushed a new version, hopefully fixing the CI issues (the tests weren't actually updated to use PER_LINUX vs. PER_LINUX32 the way the code is doing it now). At least on my system the test now runs with both "setarch x86_64" and "setarch i386". Now I really hope that the CI system agrees too..

The new version adds a new function "opinionated_personality()", a naming I must say I am particularly proud of ;-)

File: src/basic/process-util.c [hunk 2]
```diff
@@ -904,6 +904,25 @@ const char* personality_to_string(unsigned long p) {
         return architecture_to_string(architecture);
 }
 
+int opinionated_personality(unsigned long *ret) {
+        int current;
+
+        /* Returns the current personality, or PERSONALITY_INVALID if we can't determine it. This function is a bit
+         * opinionated though, and ignores all the finer-grained bits and exotic personalities, only distuingishing the
```
Comment (yuwata) [inline src/basic/process-util.c]: distuingishing -> distinguishing.

**Comment by poettering** [comment]
sorry, for dropping the ball on this one, been on vacation...

I rebased this one now, to make it apply again, and fixed the typo @yuwata pointed out. Please have another quick look & merge!

**Comment by topimiettinen** [comment]
Still looks good to me.

**Comment by poettering** [comment]
Thanks for the review! CI failure appears unrelated. Merging.

(BTW, I'll add you to the systemd commiters group shortly, if that's OK and interests you? Please feel free to merge any PRs you are sufficiently sure are good. We usually require a review by one other person before merging. Please use your new powers wisely, and if in doubt ask)

**Pull request merged**